gap/hashfun.gi

```diff
@@ -0,0 +1,383 @@
+#############################################################################
+##
+##  datastructures package
+##
+##  Copyright 2016 by the package authors.
+##  Licensed under the GPL 2 or later.
+##
+#############################################################################
+
+
+#
+
+
+#
+# Fallback method: If no other method matches, return fail.
+#
+InstallMethod( ChooseHashFunction, "fallback method",
+    [IsObject],
+function(p)
+    return fail;
+end );
+
+
+#
+# compressed GF2 vectors
+#
+InstallGlobalFunction( DATA_HashFunctionForShortGF2Vectors,
+function(v)
+    return NumberFFVector(v, 2);
+end );
+
+InstallGlobalFunction( DATA_HashFunctionForGF2Vectors,
+function(v)
+    local bytelen;
+    bytelen := QuoInt(Length(v), 8);
```

Comment:
Note that these hash functions for compressed vectors violate the assumption that equal bjects have equal hash. So they really shpuld only bes used if the domain of keys is restricted